### PR TITLE
fix: Dashboard plugin crashing UI on throw

### PR DESCRIPTION
**Pull request by AkshatJawne**
Resolves https://github.com/deephaven/web-client-ui/issues/1771

**Changes Implemented:** 
- Wrap `ErrorBoundary` around cloning of Dashboard Plugin in DashboardLayout

### CODE REVIEW

**Comment by AkshatJawne** [comment]
Will look into failures here

**Comment by AkshatJawne** [comment]
Will remove commented out code if approach works

File: packages/dashboard/src/DashboardLayout.tsx [hunk 1]
```diff
@@ -343,14 +344,16 @@ export function DashboardLayout({
       {isDashboardEmpty && emptyDashboard}
       {layoutChildren}
       {React.Children.map(children, child =>
-        child != null
-          ? React.cloneElement(child as ReactElement, {
+        child != null ? (
+          <ErrorBoundary fallback={[]}>
```
Comment (mattrunyon) [inline packages/dashboard/src/DashboardLayout.tsx:348]: Add a comment that the fallback is so we do not show the error message over the entire app and we should do something like a toast message w/ the error in the future

**Review by mattrunyon — CHANGES_REQUESTED** [review]
Update the PR body as well. Looks like what you listed was the original fix attempt

**Review by mattrunyon — APPROVED** [review]
Looks good. Also at least prevents the crash in https://github.com/deephaven/deephaven-plugins/issues/510 but that ticket still documents bad behavior in Deephaven UI, so it should stay open

**Pull request merged**